recipe/meta.yaml

```
@@ -39,6 +40,10 @@ outputs:
         - cmake-no-system  # [win]
         - ninja  # [win]
         - make  # [not win]
+      run_constrained:
+        # circular constraint on this build's xz to prevent accepting pre-split xz
+        # this is as close to 'exact' as we can get
+        - xz =={{ version }} *_{{ build }}
```

You don't need the build number, because the split was introduced with a new version (#42). So already the version constraint avoids pulling in old builds. I would just say
```suggestion
        # avoid co-installation with pre-split xz packages
        - xz =={{ version }}
```

---

The build number is required for consistency with the future, e.g. to avoid introducing the same problem again in a later refactor. The split could have happened within a version, and similar changes could happen in the future within a version, so I think this is the safest and most correct way to go (`pin_subpackage("xz", exact=True)` is certainly what _should_ go here, but doesn't work for circular dependency reasons).

FWIW, I tried `=={{ version }}[build={{ build }}]` which I believe is valid and avoids the build string glob, but conda-build mistakenly parses that matchspec as a selector comment, leading to an error.

---

> The split could have happened within a version, and similar changes could happen in the future within a version [...]

It very plainly didn't happen within a version, and we shouldn't try to prematurely optimize for a completely hypothetical split, which (if it ever happens), will be responsible for handling its own migration story.

Ultimately I'm not a maintainer on this feedstock though, so I'll leave this to y'all to hash out.

---

I'm just not sure what the downside is, when the intended constraint seems pretty clearly to be "this build of this package" which is version + build number. If there's a downside I'm happy to go with something better, I just don't see what it would be.